### PR TITLE
Fixes to `protoc` inputs to `GenerateProtoTask`

### PR DESCRIPTION
**Pull request by clayburn**
Addresses #457 and #557

### CODE REVIEW

**Review by YifeiZhuang — APPROVED** [review]

File: src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy [hunk 1]
```diff
@@ -330,12 +330,67 @@ public abstract class GenerateProtoTask extends DefaultTask {
   @Internal("Input captured by getAlternativePaths(), this is used to query alternative path by locator name.")
   abstract MapProperty<String, FileCollection> getLocatorToAlternativePathsMapping()
 
-  @InputFiles
-  @PathSensitive(PathSensitivity.NONE)
+  @Internal
+  abstract MapProperty<String, String> getLocatorToDependencyMapping()
+
+  @Internal
```
Comment (ejona86) [inline src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy:336]: This was only present for capturing input, so if it is internal it is unused. Delete?

File: src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy [hunk 2]
```diff
@@ -330,12 +330,67 @@ public abstract class GenerateProtoTask extends DefaultTask {
   @Internal("Input captured by getAlternativePaths(), this is used to query alternative path by locator name.")
```
Comment (ejona86) [inline src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy:330]: There's some references to getAlternativePaths() here and getPluginsExecutableLocators() which look to need updating.

File: src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy (hunk 1 above)
Comment (clayburn) [reply src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy:336]: Deleting certainly wouldn't break anything here, but I wasn't sure if any build scripts may be referencing the property.

Comment (ejona86) [reply src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy:336]: Oh, I see. Probably highly unlikely, but yeah, I guess leave it.

**Review by ejona86 — APPROVED** [review]

**Comment by ejona86** [comment]
@clayburn, thank you!

**Comment by Thrillpool** [comment]
This is extremely valuable for us. Will there be a release soon?

**Comment by clayburn** [comment]
@ejona86 - I've gotten a few pings about this PR. Are there any plans to make a release including this?

**Comment by ejona86** [comment]
The release is done. Unfortunately, the coordinates changed, which may cause problems for some people. Apparently just adding maven-publish plugin to the build changes the behavior of the plugin-publish plugin (https://github.com/gradle/plugin-portal-requests/issues/124). So I think I'll do another release. Release 0.8.19 is safe to use, but just annoying.

**Comment by ejona86** [comment]
Actually... I guess I won't, and will just leave it as-is. Looking at the [plugin-publish 1.0.0 release notes](https://plugins.gradle.org/plugin/com.gradle.plugin-publish/1.0.0) seems this is "the new way of doing things."

**Pull request merged**